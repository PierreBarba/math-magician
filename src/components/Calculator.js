import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    this.setState((state) => calculate(state, e.target.textContent));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <section className="containerCalculator">
        <div className="result">
          {total}
          {operation}
          {next}
        </div>
        <div className="buttonsContainer">
          <button onClick={this.handleClick} type="button" className="buttonCalculator">AC</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">+/-</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">%</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator color-right">รท</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">7</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">8</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">9</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator color-right">x</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">4</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">5</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">6</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator color-right">-</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">1</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">2</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">3</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator color-right">+</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator zero">0</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator">.</button>
          <button onClick={this.handleClick} type="button" className="buttonCalculator color-equal">=</button>
        </div>
      </section>
    );
  }
}

export default Calculator;
