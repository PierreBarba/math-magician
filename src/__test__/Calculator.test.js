import React from 'react';
import { createRoot } from 'react-dom/client';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('rendering without glitches', () => {
  const div = document.createElement('div');
  const root = createRoot(div);
  root.render(<Calculator />);
});

test('anchor Home snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});