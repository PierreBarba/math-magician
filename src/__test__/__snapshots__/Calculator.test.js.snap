// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anchor Home snapshot 1`] = `
<section
  className="containerCalculator"
>
  <div
    className="pMath"
  >
     Let's Do Some Math 
  </div>
  <div
    className="calculatorDiv"
  >
    <div
      className="result"
    />
    <div
      className="buttonsContainer"
    >
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="buttonCalculator color-right"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="buttonCalculator color-right"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="buttonCalculator color-right"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="buttonCalculator color-right"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="buttonCalculator zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="buttonCalculator"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="buttonCalculator color-equal"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
